datalad/distribution/add.py

```diff
@@ -374,7 +374,8 @@ def __call__(
                     ds.repo.add_submodule(subds_relpath, url=None, name=None)
                 except (CommandError, InvalidGitRepositoryError) as e:
                     yield get_status_dict(
-                        ds=subds, status='error', message=e.stderr,
+                        ds=subds, status='error',
+                        message=getattr(e, 'stderr', None) or str(e),
```

Comment:
s/str/six.text_type/ ?  Otherwise LGTM.

Reply:
I am ok either way since text_type is str on the strict py3 and on py2 it just does its magic, but will change so we all stay friends ;)

Reply:
> on py2 it just does its magic

That's not so obvious to me:

```python
# -*- coding: utf-8 -*-
from git.exc import InvalidGitRepositoryError
str(InvalidGitRepositoryError(u"β does not exist"))
```

```
Traceback (most recent call last):
  File "/tmp/scratch.py", line 3, in <module>
    str(InvalidGitRepositoryError(u"β does not exist"))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u03b2' in position 0: ordinal not in range(128)
```